### ISSUE TITLE
CPAN.pm + Bundle::libnet suffering severe brain damage

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#2005]
Migrated from [rt.perl.org#3280](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=3280) (status was 'resolved')

Searchable as RT3280$

**Comment by p5pRT** [comment]
### From root@erwin.omix.com

#### Created by zenin@bawdycaste.org



CPAN\.pm \+ Bundle&#8203;::libnet attempts to resolve dependencies by automatically
upgrading perl to 5\.6\.0\.  To say this is unacceptable is a severe
understatement\.

Of course\.\.\."There's a new CPAN\.pm version \(v1\.54\) available\!" and all it
says I need to do is install Bundle&#8203;::CPAN and the world should be a better
place\.\.\.until Bundle&#8203;::CPAN trys to install Bundle&#8203;::libnet and thus also trys
to upgrade perl\. \<kicks desk>

My guess is that this isn't so much a CPAN\.pm bug directly as it's seems
more likely to be bad data files \(02packages\.details\.txt\.gz\,
03modlist\.data\.gz\, etc\)?

Transcript of the CPAN shell session follows\, cut off at the point it starts
to unpack perl 5\.6\.0&#8203;:

\# perl \-MCPAN \-e shell

cpan shell \-\- CPAN exploration and modules installation \(v1\.48\)
ReadLine support available \(try \`\`install Bundle&#8203;::CPAN''\)

cpan> install Bundle&#8203;::libnet
Going to read /usr/local/cpan/sources/authors/01mailrc\.txt\.gz

&nbsp;  Please\, install Net&#8203;::FTP as soon as possible\. CPAN\.pm installs it for you
&nbsp;  if you just type
&nbsp;      install Bundle&#8203;::libnet


&nbsp;  Please\, install Net&#8203;::FTP as soon as possible\. CPAN\.pm installs it for you
&nbsp;  if you just type
&nbsp;      install Bundle&#8203;::libnet

Issuing "/usr/bin/ftp \-n"
Local directory now /usr/local/cpan/sources/modules
100% |\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*| 62727       00&#8203;:00 ETA
GOT /usr/local/cpan/sources/modules/02packages\.details\.txt\.gz
Going to read /usr/local/cpan/sources/modules/02packages\.details\.txt\.gz
Scanning cache /usr/local/cpan/build for sizes

&nbsp;  There's a new CPAN\.pm version \(v1\.54\) available\!
&nbsp;  You might want to try
&nbsp;    install Bundle&#8203;::CPAN
&nbsp;    reload cpan
&nbsp;  without quitting the current session\. It should be a seamless upgrade
&nbsp;  while we are running\.\.\.

Issuing "/usr/bin/ftp \-n"
Local directory now /usr/local/cpan/sources/modules
100% |\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*| 36849       00&#8203;:00 ETA
GOT /usr/local/cpan/sources/modules/03modlist\.data\.gz
Going to read /usr/local/cpan/sources/modules/03modlist\.data\.gz
Data&#8203;::Dumper is up to date\.
Net&#8203;::Telnet is up to date\.
Running make for G/GB/GBARR/libnet\-1\.0703\.tar\.gz

&nbsp;  CPAN&#8203;: MD5 security checks disabled because MD5 not installed\.
&nbsp;  Please consider installing the MD5 module\.

libnet\-1\.0703/
libnet\-1\.0703/Net/
libnet\-1\.0703/Net/FTP/
libnet\-1\.0703/Net/FTP/dataconn\.pm
libnet\-1\.0703/Net/FTP/L\.pm
libnet\-1\.0703/Net/FTP/A\.pm
libnet\-1\.0703/Net/FTP/E\.pm
libnet\-1\.0703/Net/FTP/I\.pm
libnet\-1\.0703/Net/POP3\.pm
libnet\-1\.0703/Net/SMTP\.pm
libnet\-1\.0703/Net/FTP\.pm
libnet\-1\.0703/Net/Cmd\.pm
libnet\-1\.0703/Net/DummyInetd\.pm
libnet\-1\.0703/Net/NNTP\.pm
libnet\-1\.0703/Net/SNPP\.pm
libnet\-1\.0703/Net/Time\.pm
libnet\-1\.0703/Net/Config\.pm
libnet\-1\.0703/Net/PH\.pm
libnet\-1\.0703/Net/Domain\.pm
libnet\-1\.0703/Net/Netrc\.pm
libnet\-1\.0703/Net/libnetFAQ\.pod
libnet\-1\.0703/demos/
libnet\-1\.0703/demos/smtp\.self
libnet\-1\.0703/demos/nntp
libnet\-1\.0703/demos/pop3
libnet\-1\.0703/demos/time
libnet\-1\.0703/demos/snpp
libnet\-1\.0703/demos/nntp\.mirror
libnet\-1\.0703/demos/ftp
libnet\-1\.0703/demos/inetd
libnet\-1\.0703/Hostname\.pm\.eg
libnet\-1\.0703/t/
libnet\-1\.0703/t/nntp\.t
libnet\-1\.0703/t/ph\.t
libnet\-1\.0703/t/require\.t
libnet\-1\.0703/t/hostname\.t
libnet\-1\.0703/t/smtp\.t
libnet\-1\.0703/t/ftp\.t
libnet\-1\.0703/libnet\.ppd
libnet\-1\.0703/README\.config
libnet\-1\.0703/MANIFEST
libnet\-1\.0703/Configure
libnet\-1\.0703/Makefile\.PL
libnet\-1\.0703/ChangeLog
libnet\-1\.0703/README
libnet\-1\.0703/Config\.eg
Removing previously used /usr/local/cpan/build/libnet\-1\.0703

&nbsp;  CPAN\.pm&#8203;: Going to build G/GB/GBARR/libnet\-1\.0703\.tar\.gz

Checking for Socket\.\.\.ok
Checking for IO&#8203;::Socket\.\.\.ok
Checking if your kit is complete\.\.\.
Looks good

Ah\, I see you already have installed libnet before\.

Do you want to modify/update your configuration \(y|n\) ? \[no\] 

This script will prompt you to enter hostnames that can be used as
defaults for some of the modules in the libnet distribution\.

To ensure that you do not enter an invalid hostname\, I can perform a
lookup on each hostname you enter\. If your internet connection is via
a dialup line then you may not want me to perform these lookups\, as
it will require you to be on\-line\.

Do you want me to perform hostname lookups \(y|n\) ? \[yes\]

The following questions all require a list of host names\, separated
with spaces\. If you do not have a host available for any of the
services\, then enter a single space\, followed by \<CR>\. To accept the
default\, hit \<CR>

Enter a list of available NNTP hosts : \[\]
Enter a list of available SMTP hosts : \[\]
Enter a list of available POP3 hosts : \[\]
Enter a list of available SNPP hosts : \[\]
Enter a list of available PH Hosts   : \[\]
Enter a list of available TIME Hosts   : \[\]
Enter a list of available DAYTIME Hosts   : \[\]

Do you have a firewall/ftp proxy  between your machine and the internet 

If you use a SOCKS firewall answer no

\(y|n\) ? \[no\]

Normally when FTP needs a data connection the client tells the server
a port to connect to\, and the server initiates a connection to the client\.

Some setups\, in particular firewall setups\, can/do not work using this
protocol\. In these situations the client must make the connection to the
server\, this is called a passive transfer\.

Should all FTP connections be passive \(y|n\) ? \[no\]

What is your local internet domain name : \[\]

If you specified some default hosts above\, it is possible for me to
do some basic tests when you run \`make test'

This will cause \`make test' to be quite a bit slower and\, if your
internet connection is via dialup\, will require you to be on\-line
unless the hosts are local\.

Do you want me to run these tests \(y|n\) ? \[yes\]

To allow Net&#8203;::FTP to be tested I will need a hostname\. This host
should allow anonymous access and have a /pub directory

What host can I use : \[\]

Writing libnet\.cfg
Writing Makefile for Net
\-\-\-\- Dependencies detected during \[G/GB/GBARR/libnet\-1\.0703\.tar\.gz\] \-\-\-\-\-
&nbsp;    Socket
&nbsp;    IO&#8203;::Socket
Running make test
&nbsp;  Make had some problems\, maybe interrupted? Won't test
Running make install
&nbsp;  Make had some problems\, maybe interrupted? Won't install
Running make for G/GS/GSAR/perl\-5\.6\.0\.tar\.gz

&nbsp;  CPAN&#8203;: MD5 security checks disabled because MD5 not installed\.
&nbsp;  Please consider installing the MD5 module\.

perl\-5\.6\.0/
perl\-5\.6\.0/AUTHORS
perl\-5\.6\.0/Artistic
perl\-5\.6\.0/Changes

&nbsp;    \.\.\.etc\, etc\.\.\.full 5\.6\.0 install\.\.\.

<details><summary>Perl Info</summary>

```


Site configuration information for perl 5.00503:

Configured by markm at Sun Mar  5 13:39:27 SAST 2000.

Summary of my perl5 (5.0 patchlevel 5 subversion 3) configuration:
  Platform:
    osname=freebsd, osvers=4.0-current, archname=i386-freebsd
    uname='FreeBSD freefall.FreeBSD.org 4.0-current FreeBSD 4.0-current #0: $Date$'
    hint=recommended, useposix=true, d_sigaction=define
    usethreads=undef useperlio=undef d_sfio=undef
  Compiler:
    cc='cc', optimize='undef', gccversion=2.95.2 19991024 (release)
    cppflags=''
    ccflags =''
    stdchar='char', d_stdstdio=undef, usevfork=true
    intsize=4, longsize=4, ptrsize=4, doublesize=8
    d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=12
    alignbytes=4, usemymalloc=n, prototype=define
  Linker and Libraries:
    ld='cc', ldflags ='-Wl,-E -lperl -lm '
    libpth=/usr/lib
    libs=-lm -lc -lcrypt
    libc=, so=so, useshrplib=true, libperl=libperl.so.3
  Dynamic Linking:
    dlsrc=dl_dlopen.xs, dlext=so, d_dlsymun=undef, ccdlflags=' -Wl,-R/usr/lib'
    cccdlflags='-DPIC -fpic', lddlflags='-Wl,-E -shared -lperl -lm '

Locally applied patches:
    


@INC for perl 5.00503:
    /usr/libdata/perl/5.00503/mach
    /usr/libdata/perl/5.00503
    /usr/local/lib/perl5/site_perl/5.005/i386-freebsd
    /usr/local/lib/perl5/site_perl/5.005
    .


Environment for perl 5.00503:
    HOME=/root
    LANG (unset)
    LANGUAGE (unset)
    LD_LIBRARY_PATH (unset)
    LOGDIR (unset)
    PATH=/sbin:/usr/sbin:/bin:/usr/bin:/usr/local/sbin:/usr/local/bin:/usr/X11R6/bin
    PERL_BADLANG (unset)
    SHELL=/bin/csh


```
</details>

**Comment by p5pRT** [comment]
### From [Unknown Contact. See original ticket]

#### Created by zenin@bawdycaste.org



CPAN\.pm \+ Bundle&#8203;::libnet attempts to resolve dependencies by automatically
upgrading perl to 5\.6\.0\.  To say this is unacceptable is a severe
understatement\.

Of course\.\.\."There's a new CPAN\.pm version \(v1\.54\) available\!" and all it
says I need to do is install Bundle&#8203;::CPAN and the world should be a better
place\.\.\.until Bundle&#8203;::CPAN trys to install Bundle&#8203;::libnet and thus also trys
to upgrade perl\. \<kicks desk>

My guess is that this isn't so much a CPAN\.pm bug directly as it's seems
more likely to be bad data files \(02packages\.details\.txt\.gz\,
03modlist\.data\.gz\, etc\)?

Transcript of the CPAN shell session follows\, cut off at the point it starts
to unpack perl 5\.6\.0&#8203;:

\# perl \-MCPAN \-e shell

cpan shell \-\- CPAN exploration and modules installation \(v1\.48\)
ReadLine support available \(try \`\`install Bundle&#8203;::CPAN''\)

cpan> install Bundle&#8203;::libnet
Going to read /usr/local/cpan/sources/authors/01mailrc\.txt\.gz

&nbsp;  Please\, install Net&#8203;::FTP as soon as possible\. CPAN\.pm installs it for you
&nbsp;  if you just type
&nbsp;      install Bundle&#8203;::libnet


&nbsp;  Please\, install Net&#8203;::FTP as soon as possible\. CPAN\.pm installs it for you
&nbsp;  if you just type
&nbsp;      install Bundle&#8203;::libnet

Issuing "/usr/bin/ftp \-n"
Local directory now /usr/local/cpan/sources/modules
100% |\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*|
62727       00&#8203;:00 ETA
GOT /usr/local/cpan/sources/modules/02packages\.details\.txt\.gz
Going to read /usr/local/cpan/sources/modules/02packages\.details\.txt\.gz
Scanning cache /usr/local/cpan/build for sizes

&nbsp;  There's a new CPAN\.pm version \(v1\.54\) available\!
&nbsp;  You might want to try
&nbsp;    install Bundle&#8203;::CPAN
&nbsp;    reload cpan
&nbsp;  without quitting the current session\. It should be a seamless upgrade
&nbsp;  while we are running\.\.\.

Issuing "/usr/bin/ftp \-n"
Local directory now /usr/local/cpan/sources/modules
100% |\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*|
36849       00&#8203;:00 ETA
GOT /usr/local/cpan/sources/modules/03modlist\.data\.gz
Going to read /usr/local/cpan/sources/modules/03modlist\.data\.gz
Data&#8203;::Dumper is up to date\.
Net&#8203;::Telnet is up to date\.
Running make for G/GB/GBARR/libnet\-1\.0703\.tar\.gz

&nbsp;  CPAN&#8203;: MD5 security checks disabled because MD5 not installed\.
&nbsp;  Please consider installing the MD5 module\.

libnet\-1\.0703/
libnet\-1\.0703/Net/
libnet\-1\.0703/Net/FTP/
libnet\-1\.0703/Net/FTP/dataconn\.pm
libnet\-1\.0703/Net/FTP/L\.pm
libnet\-1\.0703/Net/FTP/A\.pm
libnet\-1\.0703/Net/FTP/E\.pm
libnet\-1\.0703/Net/FTP/I\.pm
libnet\-1\.0703/Net/POP3\.pm
libnet\-1\.0703/Net/SMTP\.pm
libnet\-1\.0703/Net/FTP\.pm
libnet\-1\.0703/Net/Cmd\.pm
libnet\-1\.0703/Net/DummyInetd\.pm
libnet\-1\.0703/Net/NNTP\.pm
libnet\-1\.0703/Net/SNPP\.pm
libnet\-1\.0703/Net/Time\.pm
libnet\-1\.0703/Net/Config\.pm
libnet\-1\.0703/Net/PH\.pm
libnet\-1\.0703/Net/Domain\.pm
libnet\-1\.0703/Net/Netrc\.pm
libnet\-1\.0703/Net/libnetFAQ\.pod
libnet\-1\.0703/demos/
libnet\-1\.0703/demos/smtp\.self
libnet\-1\.0703/demos/nntp
libnet\-1\.0703/demos/pop3
libnet\-1\.0703/demos/time
libnet\-1\.0703/demos/snpp
libnet\-1\.0703/demos/nntp\.mirror
libnet\-1\.0703/demos/ftp
libnet\-1\.0703/demos/inetd
libnet\-1\.0703/Hostname\.pm\.eg
libnet\-1\.0703/t/
libnet\-1\.0703/t/nntp\.t
libnet\-1\.0703/t/ph\.t
libnet\-1\.0703/t/require\.t
libnet\-1\.0703/t/hostname\.t
libnet\-1\.0703/t/smtp\.t
libnet\-1\.0703/t/ftp\.t
libnet\-1\.0703/libnet\.ppd
libnet\-1\.0703/README\.config
libnet\-1\.0703/MANIFEST
libnet\-1\.0703/Configure
libnet\-1\.0703/Makefile\.PL
libnet\-1\.0703/ChangeLog
libnet\-1\.0703/README
libnet\-1\.0703/Config\.eg
Removing previously used /usr/local/cpan/build/libnet\-1\.0703

&nbsp;  CPAN\.pm&#8203;: Going to build G/GB/GBARR/libnet\-1\.0703\.tar\.gz

Checking for Socket\.\.\.ok
Checking for IO&#8203;::Socket\.\.\.ok
Checking if your kit is complete\.\.\.
Looks good

Ah\, I see you already have installed libnet before\.

Do you want to modify/update your configuration \(y|n\) ? \[no\] 

This script will prompt you to enter hostnames that can be used as
defaults for some of the modules in the libnet distribution\.

To ensure that you do not enter an invalid hostname\, I can perform a
lookup on each hostname you enter\. If your internet connection is via
a dialup line then you may not want me to perform these lookups\, as
it will require you to be on\-line\.

Do you want me to perform hostname lookups \(y|n\) ? \[yes\]

The following questions all require a list of host names\, separated
with spaces\. If you do not have a host available for any of the
services\, then enter a single space\, followed by \<CR>\. To accept the
default\, hit \<CR>

Enter a list of available NNTP hosts : \[\]
Enter a list of available SMTP hosts : \[\]
Enter a list of available POP3 hosts : \[\]
Enter a list of available SNPP hosts : \[\]
Enter a list of available PH Hosts   : \[\]
Enter a list of available TIME Hosts   : \[\]
Enter a list of available DAYTIME Hosts   : \[\]

Do you have a firewall/ftp proxy  between your machine and the internet 

If you use a SOCKS firewall answer no

\(y|n\) ? \[no\]

Normally when FTP needs a data connection the client tells the server
a port to connect to\, and the server initiates a connection to the client\.

Some setups\, in particular firewall setups\, can/do not work using this
protocol\. In these situations the client must make the connection to the
server\, this is called a passive transfer\.

Should all FTP connections be passive \(y|n\) ? \[no\]

What is your local internet domain name : \[\]

If you specified some default hosts above\, it is possible for me to
do some basic tests when you run \`make test'

This will cause \`make test' to be quite a bit slower and\, if your
internet connection is via dialup\, will require you to be on\-line
unless the hosts are local\.

Do you want me to run these tests \(y|n\) ? \[yes\]

To allow Net&#8203;::FTP to be tested I will need a hostname\. This host
should allow anonymous access and have a /pub directory

What host can I use : \[\]

Writing libnet\.cfg
Writing Makefile for Net
\-\-\-\- Dependencies detected during \[G/GB/GBARR/libnet\-1\.0703\.tar\.gz\] \-\-\-\-\-
&nbsp;    Socket
&nbsp;    IO&#8203;::Socket
Running make test
&nbsp;  Make had some problems\, maybe interrupted? Won't test
Running make install
&nbsp;  Make had some problems\, maybe interrupted? Won't install
Running make for G/GS/GSAR/perl\-5\.6\.0\.tar\.gz

&nbsp;  CPAN&#8203;: MD5 security checks disabled because MD5 not installed\.
&nbsp;  Please consider installing the MD5 module\.

perl\-5\.6\.0/
perl\-5\.6\.0/AUTHORS
perl\-5\.6\.0/Artistic
perl\-5\.6\.0/Changes

&nbsp;    \.\.\.etc\, etc\.\.\.full 5\.6\.0 install\.\.\.

<details><summary>Perl Info</summary>

```


Site configuration information for perl 5.00503:

Configured by markm at Sun Mar  5 13:39:27 SAST 2000.

Summary of my perl5 (5.0 patchlevel 5 subversion 3) configuration:
  Platform:
    osname=freebsd, osvers=4.0-current, archname=i386-freebsd
    uname='FreeBSD freefall.FreeBSD.org 4.0-current FreeBSD 4.0-current #0:
$Date$'
    hint=recommended, useposix=true, d_sigaction=define
    usethreads=undef useperlio=undef d_sfio=undef
  Compiler:
    cc='cc', optimize='undef', gccversion=2.95.2 19991024 (release)
    cppflags=''
    ccflags =''
    stdchar='char', d_stdstdio=undef, usevfork=true
    intsize=4, longsize=4, ptrsize=4, doublesize=8
    d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=12
    alignbytes=4, usemymalloc=n, prototype=define
  Linker and Libraries:
    ld='cc', ldflags ='-Wl,-E -lperl -lm '
    libpth=/usr/lib
    libs=-lm -lc -lcrypt
    libc=, so=so, useshrplib=true, libperl=libperl.so.3
  Dynamic Linking:
    dlsrc=dl_dlopen.xs, dlext=so, d_dlsymun=undef, ccdlflags=' -Wl,-R/usr/lib'
    cccdlflags='-DPIC -fpic', lddlflags='-Wl,-E -shared -lperl -lm '

Locally applied patches:
    


@INC for perl 5.00503:
    /usr/libdata/perl/5.00503/mach
    /usr/libdata/perl/5.00503
    /usr/local/lib/perl5/site_perl/5.005/i386-freebsd
    /usr/local/lib/perl5/site_perl/5.005
    .


Environment for perl 5.00503:
    HOME=/root
    LANG (unset)
    LANGUAGE (unset)
    LD_LIBRARY_PATH (unset)
    LOGDIR (unset)
   
PATH=/sbin:/usr/sbin:/bin:/usr/bin:/usr/local/sbin:/usr/local/bin:/usr/X11R6/bin
    PERL_BADLANG (unset)
    SHELL=/bin/csh


```
</details>

**Comment by p5pRT** [comment]
### From The RT System itself

Duplicate of 20000324\.024
